### PR TITLE
Swapping order of R and G LEDs, using board defined pins.

### PR DESCRIPTION
**Pull request by sandyjmacdonald**
The final hardware swaps the order of the red and green LED elements, so this PR reflects that. I've also updated the rainbow example to use `board.I2C()` when setting up the I2C bus as there are board definitions for the Pimoroni RP2040 boards in CircuitPython now!

### CODE REVIEW

**Comment by sandyjmacdonald** [comment]
How weird! It's not even the code I touched that it's failing on, as far as I can see.

**Comment by sandyjmacdonald** [comment]
Any idea why it's failing on pylint, @makermelissa ?

**Comment by makermelissa** [comment]
Yeah, PyLint was fixed recently, so some checks that were passing before are not passing now. This isn't the only repo that has been affected and we're working on it.

**Comment by tannewt** [comment]
I've proposed https://github.com/adafruit/cookiecutter-adafruit-circuitpython/pull/111 to fix the duplicate code issue. We'll need to wait for that to settle before merging (otherwise releases will fail.)

**Review by tannewt — APPROVED** [review]
Code looks good!

**Comment by sandyjmacdonald** [comment]
LOL. I got the x/y coordinates of the LEDs transposed again, because I'd flipped them in my function to map the 0-15 indices to x/y coordinates. Anyhoo... now switch 0 is 0,0, switch 1 is 1, 0, through to switch 15 being 3, 3.

**Comment by dglaude** [comment]
Maybe #32 should be considered and this follow that scheme... else the library will inflate (like with that mapping table).

**Comment by kattni** [comment]
@sandyjmacdonald Thanks for the PR! We're working on a patch to all the libraries to fix the issue causing this PR to fail. However, it may be up to a week before the patch is fully applied.

How comfortable are you with Git and GitHub? I can walk you through making the changes that need to be made to get this passing. If you're not comfortable with that, I should be able to make the changes myself and apply them to this PR. Let me know if you're interested in giving it a try!

**Comment by sandyjmacdonald** [comment]
> @sandyjmacdonald Thanks for the PR! We're working on a patch to all the libraries to fix the issue causing this PR to fail. However, it may be up to a week before the patch is fully applied.
> 
> How comfortable are you with Git and GitHub? I can walk you through making the changes that need to be made to get this passing. If you're not comfortable with that, I should be able to make the changes myself and apply them to this PR. Let me know if you're interested in giving it a try!

I can definitely give it a try!

**Comment by kattni** [comment]
@sandyjmacdonald It looks like the patch may get applied sooner than I thought, so if you'd rather simply wait, we can do that as well. However! You are welcome to try the following. It will not affect the patch as these are the changes that would be applied by the patch, and therefore the patch will simply skip this library if it's already updated.

If you would like to, what you'll do is make changes in the same branch as used for this PR (which is `master` on your fork) and `git push` the same as you did with this change. It will automatically update this PR with your new changes. If you run into issues with this, please ping me and I will help you sort it out.

Please update the following two files. 

First `Adafruit_CircuitPython_IS31FL3731/.github/workflows/build.yml` should look like the contents of [this gist](https://gist.github.com/kattni/24397716ea8abf74a1796e558ebe2fae). (You are removing the Pylint section.)

Second `Adafruit_CircuitPython_IS31FL3731/.pre-commit-config.yaml` should look like the contents of [this gist](https://gist.github.com/kattni/b98aa999cfe07ef865ee7b4b98cc06a9). (You are adding a Pylint section.)

**Comment by makermelissa** [comment]
Closing because I have included these changes as part of #38.

**Pull request closed without merging**
